tests/queries/0_stateless/03240_quantile_exact_weighted_interpolated.sql

```diff
@@ -32,4 +33,13 @@ FROM
     FROM numbers(49999)
 );
 
+SELECT 'Test with filter that returns no rows';
+SELECT medianExactWeightedInterpolated(a, 1), medianExactWeightedInterpolated(b, 2),  medianExactWeightedInterpolated(c, 3) FROM decimal WHERE a > 1000;
+
+SELECT 'Test with dynamic weights';
+SELECT medianExactWeightedInterpolated(a, w), medianExactWeightedInterpolated(b, w), medianExactWeightedInterpolated(c, w) FROM decimal;
+
+SELECT 'Test with all weights set to 0';
+SELECT medianExactWeightedInterpolated(a, 0), medianExactWeightedInterpolated(b, 0), medianExactWeightedInterpolated(c, 0) FROM decimal;
```

Comment:
This is an odd result. I'd expect weight 0 to result on a result of 0. WDYT, shouldn't it return the default value? Note that `quantileExactWeighted` is also returning odd results in this case

Reply:
Basically I would expect weight 0 to be ignored and not influence the result, the same way `-If` works:

```
:) SELECT quantileExactWeightedIf(-number - 10, (number = 42 ? 1 : 0) as w, w) FROM numbers(10);

SELECT quantileExactWeightedIf((-number) - 10, if(number = 42, 1, 0) AS w, w)
FROM numbers(10)

Query id: cd9059a6-b539-420c-8262-2bc08f19d114

   ┌─quantileExactWeightedIf(minus(negate(number), 10), w, w)─┐
1. │                                                        0 │
   └──────────────────────────────────────────────────────────┘

1 row in set. Elapsed: 0.002 sec. 

:) SELECT quantileExactWeightedIf(-number - 10, (number = 42 ? 1 : 0) as w, 1) FROM numbers(10);

SELECT quantileExactWeightedIf((-number) - 10, if(number = 42, 1, 0) AS w, 1)
FROM numbers(10)

Query id: 9f9bba04-95dc-40c3-b72a-a6bef3351868

   ┌─quantileExactWeightedIf(minus(negate(number), 10), w, 1)─┐
1. │                                                      -19 │
   └──────────────────────────────────────────────────────────┘

1 row in set. Elapsed: 0.002 sec.
```